### PR TITLE
Extract logic for bridging recognized context keys

### PR DESCRIPTION
**Pull request by anuraaga**
For #1726 figured it's good to start with making some common logic. Yet somehow the common logic takes more lines of code... I think it's still winning but not sure

### CODE REVIEW

File: gradle/enforcement/checkstyle.xml [hunk 1]
```diff
@@ -222,24 +222,24 @@
     </module>
     <module name="ClassTypeParameterName">
       <!-- modified from default Google Java Style -->
-      <property name="format" value="^[A-Z]+[0-9]?$"/>
+      <property name="format" value="(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*$)"/>
```
Comment (anuraaga) [inline gradle/enforcement/checkstyle.xml:225]: I've never quite understood the standard styles related to type parameters - a type is a type. Can we loosen to that?

**Comment by anuraaga** [comment]
With https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/3547/commits/075c27dec3b0a4cacdbc06f6ad0b0509446db1e4 I abstracted a lot more. A bit more overhead due to array / virtual method call, but guess it shouldn't be significant.

File: gradle/enforcement/checkstyle.xml (hunk 1 above)
Comment (mateuszrzeszutek) [reply gradle/enforcement/checkstyle.xml:225]: I'm fine with both CamelCase and CONSTANT_CASE, as long as you can separate words in the type name - the current MORETHANONEWORDINATYPENAME makes two- or three-word type param names really hard to read.

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by trask — APPROVED** [review]

File: instrumentation/opentelemetry-api-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/context/AgentContextStorage.java [hunk 2]
```diff
@@ -54,66 +55,19 @@
...
+        new ContextKeyBridge<Span, io.opentelemetry.api.trace.Span>(
+            "application.io.opentelemetry.api.trace.SpanContextKey",
+            "io.opentelemetry.api.trace.SpanContextKey",
+            Bridging::toApplication,
+            Bridging::toAgentOrNull),
+        new ContextKeyBridge<Baggage, io.opentelemetry.api.baggage.Baggage>(
+            "application.io.opentelemetry.api.baggage.BaggageContextKey",
+            "io.opentelemetry.api.baggage.BaggageContextKey",
+            BaggageBridging::toApplication,
+            BaggageBridging::toAgent),
```
[the remark this answers is not in the document]
Comment (anuraaga) [reply instrumentation/opentelemetry-api-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/context/AgentContextStorage.java:74]: Hmm - looked at it but not too sure how to change it so leaving it as is for now

**Pull request merged**